packages/material-ui/src/styles/createMuiTheme.js

```diff
@@ -95,6 +96,7 @@ function createMuiTheme(options = {}, ...args) {
     };
 
     traverse(muiTheme.overrides);
+    traverse(muiTheme.variants);
```

Comment:
Is this expected @oliviertassinari ?

Reply:
The traverse is meant to help catch wrong usage of the pseudo-classes. I don't think that we need to keep it. I believe it's not something we will be able to have anyway with a styled-component engine.